app/javascript/react/components/AnimalShow.js

```diff
@@ -24,40 +30,73 @@ const AnimalShow = (props) => {
         const errorMessage = `${response.status} (${response.statusText})`
         throw new Error(errorMessage)
       }
+
       const responseBody = await response.json()
       setAnimal({
         ...responseBody.animal,
-        ["reviews"]: responseBody.animal.reviews,
+        ["reviews"]: responseBody.animal.reviews
+      })
+      setCurrentUser({
+        username: responseBody.animal.current_user.username,
+        profile_photo: responseBody.animal.current_user.profile_photo,
+        role: responseBody.animal.current_user.role
       })
-      setUserId(responseBody.animal.current_user.id)
     } catch (error) {
       console.error(`Error in Fetch: ${error.message}`)
     }
   }
 
-  const postReview = async (formPayload) => {
+  const postReview = async formPayload => {
     try {
       let animalId = props.match.params.id
       const response = await fetch(`/api/v1/animals/${animalId}/reviews`, {
         credentials: "same-origin",
         method: "POST",
         headers: {
           Accept: "application/json",
-          "Content-Type": "application/json",
+          "Content-Type": "application/json"
         },
-        body: JSON.stringify(formPayload),
+        body: JSON.stringify(formPayload)
       })
       if (!response.ok) {
         const errorMessage = `${response.status} (${response.statusText})`
         throw new Error(errorMessage)
       }
       const newReview = await response.json()
-      setAnimal({ ...animal, ["reviews"]: animal.reviews.concat(newReview) })
+      let animalReviewsList = animal.reviews
+      animalReviewsList = animalReviewsList.concat(newReview.review)
+      setAnimal({
+        ...animal,
+        ["reviews"]: animalReviewsList
+      })
     } catch (error) {
       console.error(`Error in Fetch: ${error.message}`)
     }
   }
 
+  const deleteReview = async reviewId => {
+    try {
+      const response = await fetch(
+        `/api/v1/animals/${animal.id}/reviews/${reviewId}`,
+        {
+          credentials: "same-origin",
+          method: "DELETE",
+          headers: {
+            Accept: "application/json",
+            "Content-Type": "application/json"
+          }
+        }
+      )
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`
+        throw new Error(errorMessage)
+      }
+      fetchAnimal()
```

Comment:
So instead of running a second fetch request, we can actually have our controller send back an updated list of reviews (minus the deleted one). That can then update the state.